src/charm.py

```diff
@@ -49,17 +49,30 @@ def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR
         ...
 
 
+class LimitToOneRequest:
+    """Filter the CSR so as to only allow a single request from any relation ID."""
+
+    def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR]) -> bool:
+        """Accept CSR if its the first CSR of a relation or the renewal of the existing CSR."""
+        relevant_csrs = [csr for csr in requirer_csrs if csr.relation_id == relation_id]
+        if len(relevant_csrs) > 1:
+            return False
```

Comment:
I think logging why we are denying the certificate here would be helpful, and would fit with the log I added in the event handler.